### PR TITLE
Feature/Refactor Dashboard Layout

### PR DESCRIPTION
**Pull request by johnedma**
Soundtrack of this PR: [James Harden & Russell Westbrook - Digital Dash - Drake & Future](https://www.youtube.com/watch?v=4lJq_JtJUuQ)

### Motivation

Continued DW refactoring, adding types and testing to components.

### In this PR

- DashboardLayout
- BalanceIndicator
- InactivityDetect
- OnboardingModal
- SetPinModal
- ShowEntropyModal
- SyncStatus
- TopBar/NavBar

### Future Work

- Additional testing, stories and clean up.

### Testing

- Test Suites: 7 passed, 7 total
Tests:       16 passed, 16 total
Snapshots:   0 total
Time:        7.225 s
Ran all test suites matching /app\/layouts\/DashboardLayout/i.

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -26,22 +26,23 @@ Soundtrack of this PR: [link to song that really fits the mood of this PR]()
...
-  axiosFullService.ts |     100 |      100 |     100 |     100 |
- fullService/api      |     100 |      100 |     100 |     100 |
-  getLockedStatus.ts  |     100 |      100 |     100 |     100 |
-  index.ts            |       0 |        0 |       0 |       0 |
+All files | 100 | 100 | 100 | 100 |
+fullService | 100 | 100 | 100 | 100 |
+axiosFullService.ts | 100 | 100 | 100 | 100 |
+fullService/api | 100 | 100 | 100 | 100 |
+getLockedStatus.ts | 100 | 100 | 100 | 100 |
+index.ts | 0 | 0 | 0 | 0 |
```
Comment (fktt22) [inline .github/pull_request_template.md:43]: Why this change? I think this file should be removed from the PR.

File: app/layouts/DashboardLayout/BalanceIndicator.view/BalanceIndicator.d.ts [hunk 2]
```diff
@@ -0,0 +1,4 @@
+export interface BalanceIndicatorProps {
+  balance: string | '';
```
Comment (fktt22) [inline app/layouts/DashboardLayout/BalanceIndicator.view/BalanceIndicator.d.ts:2]: Why the `| ''` part? It's still a string.

File: app/layouts/DashboardLayout/BalanceIndicator.view/BalanceIndicator.test.tsx [hunk 3]
```diff
@@ -1,20 +1,19 @@
...
 
-import BalanceIndicator from '../../../../app/views/wallet/DashboardView/BalanceIndicator';
+import { BalanceIndicator } from './BalanceIndicator.view';
 
-const MOCK_BALANCE = '1234230000000';
+const MOCK_BALANCE = '10.000000000000';
 
 describe('BalanceIndicator', () => {
   test('renders the correct balance', () => {
     render(<BalanceIndicator balance={MOCK_BALANCE} isSynced />);
```
Comment (fktt22) [inline app/layouts/DashboardLayout/BalanceIndicator.view/BalanceIndicator.test.tsx:13]: I think we also need a test showing the "unsynced" message if `isSynced===false`

File: app/layouts/DashboardLayout/InactivityDetect/InactivityDetect.test.tsx [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+    accountBlockIndex: '1234',
+    isSynced: true,
+    localBlockIndex: '1234',
+    networkBlockIndex: '1234',
+    object: 'balance',
+    orphanedPmob: '1234',
+    pendingPmob: '1234',
+    secretedPmob: '1234',
+    spentPmob: '1234',
+    unspentPmob: '1234',
```
Comment (fktt22) [inline app/layouts/DashboardLayout/InactivityDetect/InactivityDetect.test.tsx:30]: Just a tip, though it doesn't affect here: numbers and strings should all be different, to make sure that the right one is displayed. Here it makes no difference because there are no tests for 1234 appearing anywhere, but it's a good practice anyway.

File: app/layouts/DashboardLayout/InactivityDetect/InactivityDetect.test.tsx [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+  },
+};
+
+const TIME_FOR_INACTIVITY = 0;
```
Comment (fktt22) [inline app/layouts/DashboardLayout/InactivityDetect/InactivityDetect.test.tsx:34]: Nice!

File: app/layouts/DashboardLayout/InactivityDetect/InactivityDetect.view.tsx [hunk 6]
```diff
@@ -54,7 +57,7 @@ const InactivityDetect: FC = () => {
       setInactiveTooLong(true);
       document.onmousemove = null;
       document.onkeypress = null;
-      reactionTimer = window.setTimeout(() => window.close(), TIME_FOR_REACTION);
+      reactionTimer = window.setTimeout(() => handleCloseApp(), TIME_FOR_REACTION);
```
Comment (fktt22) [inline app/layouts/DashboardLayout/InactivityDetect/InactivityDetect.view.tsx:60]: This can also be written: `setTimeout(handleCloseApp, TIME_FOR_REACTION)`

File: app/layouts/DashboardLayout/OnboardingModal.view/OnboardingModal.test.tsx [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+        isEntropyKnown={false}
+        isPinRequired
+      />
+    );
+
+    expect(getByText(showEntropyMsg)).toBeInTheDocument();
+  });
+
+  test('renders SetPinModal when pin is required', () => {
+    const { getByText } = render(
```
Comment (fktt22) [inline app/layouts/DashboardLayout/OnboardingModal.view/OnboardingModal.test.tsx:33]: The render logic --which is repeated several times-- could be refactored to a function.

File: app/layouts/DashboardLayout/SetPinModal.view/SetPinModal.test.tsx [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+      selector: 'input',
+    }) as HTMLInputElement;
+    const pinThresholdField = screen.getByLabelText('Minimum amount that requires a PIN', {
+      selector: 'input',
+    }) as HTMLInputElement;
+    const submitButton = screen.getByRole('button', { name: 'Set PIN' });
+
+    userEvent.type(newPinField, pin);
+    userEvent.type(confirmPinField, pin);
+    userEvent.type(pinThresholdField, pinThreshold);
```
Comment (fktt22) [inline app/layouts/DashboardLayout/SetPinModal.view/SetPinModal.test.tsx:42]: Isn't act(...) needed here? Formik needs it.

File: app/layouts/DashboardLayout/SetPinModal.view/SetPinModal.test.tsx [hunk 9]
```diff
@@ -0,0 +1,78 @@
...
```
Comment (fktt22) [inline app/layouts/DashboardLayout/SetPinModal.view/SetPinModal.test.tsx:70]: Same question here

File: app/layouts/DashboardLayout/SyncStatus.view/SyncStatus.TEST.tsx [hunk 10]
```diff
@@ -0,0 +1,109 @@
...
+        accountBlockIndex: '1234',
+        isSynced: true,
+        localBlockIndex: '1234',
+        networkBlockIndex: '123',
+        object: 'balance',
+        orphanedPmob: '1234',
+        pendingPmob: '1234',
+        secretedPmob: '1234',
+        spentPmob: '1234',
+        unspentPmob: '1234',
```
Comment (fktt22) [inline app/layouts/DashboardLayout/SyncStatus.view/SyncStatus.TEST.tsx:32]: See earlier comment about repeating values

File: app/layouts/DashboardLayout/SyncStatus.view/SyncStatus.TEST.tsx [hunk 11]
```diff
@@ -0,0 +1,109 @@
...
+        isSynced: true,
+        localBlockIndex: '1234',
+        networkBlockIndex: '1234',
+        object: 'balance',
+        orphanedPmob: '1234',
+        pendingPmob: '1234',
+        secretedPmob: '1234',
+        spentPmob: '1234',
+        unspentPmob: '1234',
+      },
```
Comment (fktt22) [inline app/layouts/DashboardLayout/SyncStatus.view/SyncStatus.TEST.tsx:99]: Couldn't this be defined as a const? Is it the same always, in all tests?

**Review by fktt22 — DISMISSED** [review]
Please see comments -- several minor, and a couple of suggestions

File: .github/pull_request_template.md (hunk 1 above)
Comment (johnedma) [reply .github/pull_request_template.md:43]: I made an edit to the original when starting this branch with gitflow to try and have it use the info I put into the template, didn't work so this it to restore it to its original state.

File: app/layouts/DashboardLayout/BalanceIndicator.view/BalanceIndicator.d.ts (hunk 2 above)
Comment (johnedma) [reply app/layouts/DashboardLayout/BalanceIndicator.view/BalanceIndicator.d.ts:2]: Someone else wrote this particular type, I just exported the interface. Wasn't sure (and still not) if there's a scenario where removing this would cause a user not to be able to see there balance.

File: app/layouts/DashboardLayout/BalanceIndicator.view/BalanceIndicator.test.tsx (hunk 3 above)
Comment (johnedma) [reply app/layouts/DashboardLayout/BalanceIndicator.view/BalanceIndicator.test.tsx:13]: Thats on line 25.

File: app/layouts/DashboardLayout/OnboardingModal.view/OnboardingModal.test.tsx (hunk 7 above)
Comment (johnedma) [reply app/layouts/DashboardLayout/OnboardingModal.view/OnboardingModal.test.tsx:33]: It's called 3 times, with a varying property on each call - I think it's ok.

File: app/layouts/DashboardLayout/SetPinModal.view/SetPinModal.test.tsx (hunk 8 above)
Comment (johnedma) [reply app/layouts/DashboardLayout/SetPinModal.view/SetPinModal.test.tsx:42]: Nope, it works. On line 50 theres a test to throw an error for an invalid entry, which means it had to actually receive the input to know when to throw the error and when not.

File: app/layouts/DashboardLayout/SyncStatus.view/SyncStatus.TEST.tsx (hunk 11 above)
Comment (johnedma) [reply app/layouts/DashboardLayout/SyncStatus.view/SyncStatus.TEST.tsx:99]: The values aren't the same, read the actual description in 'test' to see the properties being manipulated that make the difference for that particular test.

File: app/layouts/DashboardLayout/BalanceIndicator.view/BalanceIndicator.d.ts (hunk 2 above)
Comment (BrianCorbin) [reply app/layouts/DashboardLayout/BalanceIndicator.view/BalanceIndicator.d.ts:2]: Should be covered in a test, let's just remove the | '' part and move forward :)

File: app/layouts/DashboardLayout/SyncStatus.view/SyncStatus.view.tsx [hunk 12]
```diff
@@ -82,7 +80,7 @@ const SyncStatus: FC = () => {
 
     statusCode = isSynced ? SYNCED : SYNCING;
   }
-  ipcRenderer.send('sync-status', statusCode);
+  sendSyncStatus(statusCode);
```
Comment (BrianCorbin) [inline app/layouts/DashboardLayout/SyncStatus.view/SyncStatus.view.tsx:83]: How come we're sending a status code up, is this not something that should be managed further up the chain?

Comment (johnedma) [reply app/layouts/DashboardLayout/SyncStatus.view/SyncStatus.view.tsx:83]: It may appear as if its being passed up but it's actually inside DashboardLayout which wraps every route rendered. Once the user is authenticated SyncStatus runs before even displaying the component so I'm not sure of a better location.

File: app/layouts/DashboardLayout/BalanceIndicator.view/BalanceIndicator.d.ts (hunk 2 above)
Comment (johnedma) [reply app/layouts/DashboardLayout/BalanceIndicator.view/BalanceIndicator.d.ts:2]: 👍🏾

Comment (fktt22) [reply app/layouts/DashboardLayout/BalanceIndicator.view/BalanceIndicator.d.ts:2]: Agree: remove the ''

File: app/layouts/DashboardLayout/OnboardingModal.view/OnboardingModal.test.tsx (hunk 7 above)
Comment (fktt22) [reply app/layouts/DashboardLayout/OnboardingModal.view/OnboardingModal.test.tsx:33]: Pass the different value as a parameter to the function, and that will solve it.

File: app/layouts/DashboardLayout/SetPinModal.view/SetPinModal.test.tsx (hunk 8 above)
Comment (fktt22) [reply app/layouts/DashboardLayout/SetPinModal.view/SetPinModal.test.tsx:42]: Weird...
